app.js

```diff
@@ -180,34 +184,55 @@
         }
     });
 
-    app.post("/login", function(req, res) {
+app.post("/login", async function (req, res, next) {
+    try {
+
+
         const user = new User({
             username: req.body.username,
             password: req.body.password
         });
-        User.findOne({
-            username: req.body.username
-        }, function(err, found) {
-            if (!found) {
-                var message = "No such username exists";
-                res.render('login', {
-                    message: message
-                });
-            } else {
-                // console.log(req.body);
-                req.login(user, function(err) {
-                    if (err) {
-                        console.log(err);
-                    } else {
-                        passport.authenticate("local")(req, res, function() {
+    
+   
+        const found = await User.findOne({ username: req.body.username })
+        if (!found) {
+            var message = "No such username exists";
+           return res.render('login', {
+                message: message
+            });
+        } 
+          
+        passport.authenticate("local",
+            (err, user, options) => {
+                console.log(err,user)
+                if (user) {
+                    
+                    req.login(user, (error) => {
+                        if (error) {
+                           res.render("login",{
+                            message:error.message
+                        })
+                          
+                        } else {
+
                             if (req.body.username === 'admin') res.redirect("/AdminHome");
-                            else res.redirect("/UserHome");
-                        });
-                    }
-                });
+                                                        else res.redirect("/UserHome");
+                        };
+                    });
+                }
+                else {
+                    res.render("login", {
```

Comment:
here if req.body.username === admin then render the `adminlogin` page
otherwise `login` page